lib/module.js

```diff
@@ -434,7 +435,34 @@ Module._load = function(request, parent, isMain) {
     debug('Module._load REQUEST %s parent: %s', request, parent.id);
   }
 
-  var filename = Module._resolveFilename(request, parent, isMain);
+  var filename = null;
+
+  if (isMain) {
+    let err;
+    try {
+      filename = Module._resolveFilename(request, parent, isMain);
+    } catch (e) {
+      // try to keep stack
+      e.stack;
+      err = e;
+    }
+    if (filename === null || /\.mjs$/.test(filename)) {
+      try {
+        ESM.loader.import(request).catch((e) => {
+          console.error(e);
+          process.exit(1);
+        });
+        return;
```

Comment:
There are weird things if you make `require` work that I'd rather not go into great depth here about. For now `require("esm")` does nothing new.